crates/hir-ty/src/infer/expr.rs

```diff
@@ -334,7 +334,7 @@ impl InferenceContext<'_> {
                     ExprIsRead::No
                 };
                 let input_ty = self.infer_expr(expr, &Expectation::none(), child_is_read);
-                self.infer_top_pat(pat, &input_ty);
+                self.infer_top_pat(pat, &input_ty, None);
```

Comment:
While we have no use for it now, we should put a `Some(DeclContext { origin: DeclOrigin::LetExpr }` here